torch/csrc/distributed/rpc/init.cpp

```diff
@@ -23,6 +23,8 @@ namespace rpc {
 
 namespace {
 
+constexpr std::chrono::milliseconds kDeleteAllUsersTimeout(100000);
```

Comment:
This is looks good to me for now. In future PRs we probably would like to expose this to users for configure?

Reply:
Yes, for now I prefer not exposing to shutdown API this moment.

Because I think users should only care to pass a top-level timeout, that includes 
0) wait_for_all_workers, to flush any user RPC requests.
1) wait for pending, 
2) send and wait delete user forks, 
3) wait for owners to be removed. 
4) wait for join().

This will require `utils::Future` to take timeout on it's wait() function, and require RpcAgent sync()/join() to take timeout as well.